### PR TITLE
feat: live update primitive field values

### PR DESCRIPTION
**Pull request by z0al**
Todo:
- [ ] Fetch Content types from Host on the initial load
- [ ] More testing

### CODE REVIEW

File: src/graphql/entries.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import { isPrimitiveField, logUnrecognizedFields } from './utils';
+
+/**
+ * Updates GraphQL response data based on CMA entry object
+ *
+ * @param contentType entity
+ * @param data the GraphQL response to be updated
+ * @param update CMA entry object containing the update
+ * @param locale locale code
```
Comment (YvesRijckaert) [inline src/graphql/entries.ts:11]:
```suggestion
 * @param locale code
```

**Review by YvesRijckaert — APPROVED** [review]
💯

File: src/live-updates.ts [hunk 2]
```diff
@@ -16,9 +17,11 @@ export class LiveUpdates {
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   private mergeGraphQL(initial: Argument, locale: string, updated: Entity): Argument {
-    // TODO: https://contentful.atlassian.net/browse/TOL-1000
-    // TODO: https://contentful.atlassian.net/browse/TOL-1022
-    return initial;
+    if ((initial as any).__typename === 'Asset') {
```
Comment (chrishelgert) [inline src/live-updates.ts:20]: `Argument` can also be an array currently, maybe we should adjust this check here

File: src/graphql/utils.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  for (const field of Object.keys(data)) {
+    if (!recognized.has(field)) {
+      console.warn(`Unrecognized field '${field}'. Note that GraphQL aliases are not supported`);
+    }
+  }
+}
+
+export function isPrimitiveField(field: ContentFields): boolean {
+  const types = new Set(['Symbol', 'Text', 'Integer', 'Boolean', 'Date', 'Location', 'Object']);
```
Comment (aodhagan-cf) [inline src/graphql/utils.ts:17]: nit: I would extract this out of the function so you don't have to instantiate it even function call

File: src/graphql/entries.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    fields.map((f) => f.apiName ?? f.name),
+    data
+  );
+
+  for (const field of fields) {
+    const name = field.apiName ?? field.name;
+
+    if (isPrimitiveField(field) && name in data) {
+      // Falling back to 'null' as it's what GraphQL users would expect
+      // FIXME: handle locale fallbacks
```
Comment (aodhagan-cf) [inline src/graphql/entries.ts:33]: Would also be nice to capture the fallback in a test

File: src/live-updates.ts (hunk 2 above)
Comment (aodhagan-cf) [reply src/live-updates.ts:20]: Btw I think they handle the type in the starter template like this

```
{
    sys: { id: string };
    __typename: string;
    [k: string]: any;
}
```

**Pull request merged**